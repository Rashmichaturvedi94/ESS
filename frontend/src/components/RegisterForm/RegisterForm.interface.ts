import { FormProps } from '../../hooks/useForm.interface';

export interface RegisterFormValues {
  name: string;
  username: string;
  email: string;
  password: string;
  password2: string;
  is_publisher?: boolean;
}

export type RegisterFormProps = FormProps<RegisterFormValues>;
