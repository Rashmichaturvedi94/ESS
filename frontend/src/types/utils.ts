export type Nullable<T> = {
  [P in keyof T]: T[P] | null;
};

export type Uri = string;

export type DateString = string;

export type LocalOrRemoteFile = File | Uri;

export type Modify<T, R> = Omit<T, keyof R> & R;
