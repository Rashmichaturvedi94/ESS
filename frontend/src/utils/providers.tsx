import React, { ComponentType, FC } from 'react';
import * as Sentry from '@sentry/react';
import { QueryClientProvider } from 'react-query';
import DateFnsUtils from '@date-io/date-fns';
import { ThemeProvider } from 'styled-components';
import { StylesProvider, MuiThemeProvider } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { queryClient } from '../api';
import { theme } from '../styles/theme';


const createProviderHOC = (Provider: ComponentType) => <P extends object>(
  Component: ComponentType<P>,
) => (props: P) => (
  <Provider>
    <Component {...props} />
  </Provider>
);
const FallbackComponent = () => <div>An error has occurred</div>;

export const Providers: FC = ({ children }) => (
  <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
    <StylesProvider injectFirst>
      <MuiThemeProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <QueryClientProvider client={queryClient}>
              {children}
            </QueryClientProvider>
          </MuiPickersUtilsProvider>
        </ThemeProvider>
      </MuiThemeProvider>
    </StylesProvider>
  </Sentry.ErrorBoundary>
);

export const withProviders = createProviderHOC(Providers);
