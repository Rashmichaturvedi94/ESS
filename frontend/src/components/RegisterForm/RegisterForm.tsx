import React, { FC } from 'react';
import { Box, FormHelperText, Radio, RadioGroup } from '@material-ui/core';
import Button from '@material-ui/core/Button';
import * as Yup from 'yup';
import { NameField, UsernameField, EmailField, PasswordField, RadioField } from './RegisterForm.styles';
import { useForm } from '../../hooks/useForm';
import { RegisterFormProps, RegisterFormValues } from './RegisterForm.interface';

const initialValues: RegisterFormValues = {
  name: '',
  username: '',
  email: '',
  password: '',
  password2: '',
  is_publisher: false,
};

const validationSchema: Yup.SchemaOf<RegisterFormValues> = Yup.object().shape({
  name: Yup.string().required('Required field'),
  username: Yup.string().required('Required field'),
  email: Yup.string().required('Required field'),
  password: Yup.string().required('Required field'),
  password2: Yup.string().required('Required field'),
  is_publisher: Yup.boolean().required('Required field'),
});

export const RegisterForm: FC<RegisterFormProps> = (props) => {
  const { errors, values, handleChange, handleSubmit } = useForm({
    initialValues,
    validationSchema,
    ...props,
  });

  const [value, setValue] = React.useState('');

  const handleRadioChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setValue((event.target as HTMLInputElement).value);
  };

  // eslint-disable-next-line no-return-assign
  return (
    <form onSubmit={handleSubmit}>
      <Box display="flex" flexDirection="column" gridRowGap={20} style={{ margin: '0px 220px'}}>
        {errors?.detail && (
          <FormHelperText error>{errors?.detail}</FormHelperText>
        )}
        <NameField
          fullWidth
          name="name"
          placeholder='Name'
          error={!!errors.name}
          helperText={errors.name}
          value={values.name}
          onChange={handleChange}
        />
        <UsernameField
          fullWidth
          name="username"
          placeholder='Username'
          error={!!errors.username}
          helperText={errors.username}
          value={values.username}
          onChange={handleChange}
        />
        <EmailField
          fullWidth
          name="email"
          placeholder='Email'
          error={!!errors.email}
          helperText={errors.email}
          value={values.email}
          onChange={handleChange}
        />
        <PasswordField
          fullWidth
          name="password"
          type="password"
          placeholder='Password'
          error={!!errors.password}
          helperText={errors.password}
          value={values.password}
          onChange={handleChange}
        />
        <PasswordField
          fullWidth
          name="password2"
          type="password2"
          placeholder='Confirm Password'
          error={!!errors.password2}
          helperText={errors.password2}
          value={values.password2}
          onChange={handleChange}
        />
          <RadioGroup aria-label="Type" name="is_publisher" value={value} onChange={handleRadioChange}>
            <RadioField value="true" control={<Radio color="default" />} label="Author"/>
            <RadioField value="false" control={<Radio color="default"/>} label="Student" />
          </RadioGroup>
        <Button type="submit" variant="contained" fullWidth>
          Register
        </Button>
      </Box>
    </form>
  );
};
