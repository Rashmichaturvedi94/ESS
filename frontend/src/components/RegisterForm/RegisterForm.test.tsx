import React from 'react';
import { render } from '../../utils/testUtils';
import { RegisterForm } from './RegisterForm';

describe('components/RegisterForm', () => {
  it('should render', () => {
    const { container } = render(<RegisterForm />);
    expect(container).toMatchSnapshot();
  });
});
