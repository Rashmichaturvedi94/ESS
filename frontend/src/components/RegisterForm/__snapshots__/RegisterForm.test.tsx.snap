// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RegisterForm should render 1`] = `
<div>
  <form>
    <div
      class="MuiBox-root MuiBox-root-1"
      style="margin: 0px 220px;"
    >
      <div
        class="MuiFormControl-root MuiTextField-root sc-gtsrHT bdAwyA MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            name="name"
            placeholder="Name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root sc-dlnjwi cqBiuT MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            name="username"
            placeholder="Username"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root sc-hKFxyN GcSgM MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            name="email"
            placeholder="Email"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root sc-eCApnc cyyBMT MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root sc-eCApnc cyyBMT MuiFormControl-fullWidth"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            name="password2"
            placeholder="Confirm Password"
            type="password2"
            value=""
          />
        </div>
      </div>
      <div
        aria-label="Type"
        class="MuiFormGroup-root"
        role="radiogroup"
      >
        <label
          class="MuiFormControlLabel-root sc-jSFjdj bBwAyG"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiRadio-root"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-5"
                name="Type1"
                type="radio"
                value="Author"
              />
              <div
                class="PrivateRadioButtonIcon-root-6"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-7"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Author
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root sc-jSFjdj bBwAyG"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiRadio-root"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-5"
                name="Type1"
                type="radio"
                value="Student"
              />
              <div
                class="PrivateRadioButtonIcon-root-6"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-7"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Student
          </span>
        </label>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Register
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
